### ISSUE TITLE
AG News Dataset

### ISSUE THREAD

**Issue by jrohsc** [thunlp/ONION#3]
Hi, I was just wondering if I can obtain model parameters and the poison dataset for AG News as shown in the paper. 

I tried to generate the poisoned AG News dataset for BadNet by myself (5 random rare word triggers in random position of the sentence) but both the ASR and the CACC of the BERT model remained around 88% to 90% during the testing phase, while the paper suggests an ASR: 100% and CACC: 93.97%.  Is the model parameter same as the model for SST-2 and offenseval?

Screenshot of the Training:

<img width="970" alt="AG News Badnet Attack" src="https://user-images.githubusercontent.com/56494297/166207445-4572c1e5-61ed-4d70-9d4c-b0392864d998.png">

I was also wondering if I can get the dataset for BN(m) and BN(h) if possible. 

I will be waiting for the reply :)

**Comment by Yangyi-Chen** [comment]
Hi, thanks for watching! 

I've uploaded BN(m) & BN(h) & Badnets to [Google Drive](https://drive.google.com/file/d/1lbCC3RAr4w5ge3n9gpsfpRLgWgAEmmVZ/view?usp=sharing). It will be available for around 5 days. Thanks! 

And actually, I don't know what you mean by "Is the model parameter the same as the model for SST-2 and offenseval". We train these models on different poison datasets so the parameters are definitely different. 

For the lower attack success rate, maybe the training paradigm should be adjusted. We follow the standard hyper-parameters setting (our default implementation). 

If it still doesn't work, pls feel free to let me know. And it would be great if you can provide more detailed information (e.g. hyper-parameters setting, et al). 

Can directly find me at: yangyichen6666@gmail.com

**Comment by jrohsc** [comment]
> 

Thank you for the reply! 

For this "Is the model parameter the same as the model for SST-2 and offenseval": I was referring to the hyperparameter for the model setting.

**Comment by Yangyi-Chen** [comment]
I see! 

The hyper-parameters setting:
poison_rate: 5
batch_size:32
optimizer: adam
epoch: 3 warm_up_epochs, 5 normal epochs.
lr: 2e-5
scheduler: linear decay 
weight_decay: 0

I think the hyper-parameters setting is consistent across three datasets, and we don't observe too much difference in the attack performance. 

Hope this helps, thanks!
